### PR TITLE
AG-7585 Treemap API improvements & docs

### PR DESCRIPTION
**Pull request by alexanderby**
- Untie tile and tooltip label text from colorKey.
- Ability to get depth and parent in formatter and tooltip renderer.
- Separate tile and text shadows.
- Highlight text colour (can be used in all types).
- Ability to prevent groups highlighting.
- Fixed not displaying formatted fill in the tooltip.
- API docs update and complex colouring example.

### CODE REVIEW

File: charts-packages/ag-charts-community/src/chart/series/series.ts [hunk 1]
```diff
@@ -95,9 +95,15 @@ class SeriesHighlightStyle {
...
 
+class TextHighlightStyle {
+    @Validate(OPT_COLOR_STRING)
+    color?: string = 'black';
+}
+
 export class HighlightStyle {
     readonly item = new SeriesItemHighlightStyle();
     readonly series = new SeriesHighlightStyle();
+    readonly text = new TextHighlightStyle();
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/chart/series/series.ts:106]: I don't see any update for cartesian series types related to this addition? (Importantly, the `AgSeriesHighlightStyle` change means that docs for cartesian series types will include this options - so we should either rectify that or add support in those other series types?)

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-treemap-series/examples/custom-colors/main.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                },
+            },
+            formatter: ({ datum, depth, parent, highlighted }) => {
+                if (depth < 2) {
+                    return {};
+                }
+
+                const stroke = highlighted ? 'black' : 'white';
+                const mainColor = parent.name === 'Foodstuffs' ? [224, 64, 32] : [32, 92, 224];
+                const siblings = parent.children.sort((a, b) => b.exports - a.exports);
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-treemap-series/examples/custom-colors/main.ts:47]: `Array.sort()` mutates the array - this raises an interesting point regarding mutability of the callback parameters here?

**Review by alantreadway — COMMENTED** [review]
Doc updates look great, just a couple of things to follow-up on.

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-treemap-series/examples/custom-colors/main.ts (hunk 2 above)
Comment (alexanderby) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-treemap-series/examples/custom-colors/main.ts:47]: Yes, probably the coloring is unnecessary complex here, I will make just 2 plain colors for different branches.

File: charts-packages/ag-charts-community/src/chart/series/series.ts (hunk 1 above)
Comment (alexanderby) [reply charts-packages/ag-charts-community/src/chart/series/series.ts:106]: Will do just for Treemap. It could also be useful for Pies.

**Review by alantreadway — APPROVED** [review]
LGTM - thanks for the updates!

**Pull request merged**